index.html

```diff
@@ -0,0 +1,482 @@
+      }
+    }
+  }
+  
+}
+```
+
+`Astronaut.name` is provided by the `astronauts` subgraph. But the `captain` field is provided by the `rockets` subgraph.
+
+The fields specified in `repr` will be passed to the subgraph's `Query._entities(representations:)` as an item within the `representations` list.
+
+Multiple `@csdl_key`s can be provided for different graphs, or for the same graph.
+
+```graphql
+#::[Using `@csdl_key` to specify subgraph keys]
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{ ... }
+```
+
+### `@csdl_endpoint`
+
+```graphql
+#::[`@csdl_endpoint` declaration]
+directive @csdl_endpoint(url: csdl_URL) on ENUM_VALUE
+```
+
+Bind an endpoint URL to a subgraph. This directive is only valid on enum values within the required `csdl_Graph` enum type.
+
+```graphql
+#::[Using `@csdl_endpoint` to specify subgraph endpoints]
+enum csdl_Graph {
+  ROCKETS @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+```
+
+### `@csdl_resolve`
+
+```graphql
+#::[`@csdl_resolve` declaration]
+directive @csdl_resolve(
+  graph: csdl_Graph!,
+  requires: csdl_SelectionSet,
+  provides: csdl_SelectionSet) on FIELD_DEFINITION
+```
+
+Bind a subgraph resolver to this field.
+
+Any field definitions without a `@csdl_resolve` directive are *free*. That is, the CSDL asserts they can be resolved by any subgraph in which the parent type can be found. Specifying `@csdl_resolve` binds a field to resolve in exactly one subgraph. Unless it is a root type, the enclosing type **must** be [portable](#portability) to the specified subgraph (it must have `@csdl_key`s specified for that graph).
+
+```graphql
+#::[Using `@csdl_resolve` to specify subgraph resolvers]
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, fields: "{ id }")
+  @csdl_key(graph: ROCKETS, fields: "{ id }")
+{
+  id: String!     @csdl_resolve(graph: ASTRONAUTS)
+  name: String!   @csdl_resolve(graph: ASTRONAUTS)
+  tripId: String! @csdl_resolve(graph: ASTRONAUTS)
+  rocket: Rocket! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Rocket 
+  @csdl_key(fields: "{ id }", graph: ASTRONAUTS)
+  @csdl_key(fields: "{ id }", graph: ROCKETS)
+{
+  id: String!           @csdl_resolve(graph: ROCKETS)
+  name: String!         @csdl_resolve(graph: ROCKETS)
+  astronaut: Astronaut! @csdl_resolve(graph: ASTRONAUTS)
+  captain: Astronaut    @csdl_resolve(graph: ASTRONAUTS,
+                                      provides: "{ tripId }")
+}
+```
+
+Fields on root types must always be bound to a subgraph:
+
+```graphql
+#::[`@csdl_resolve` on root type fields]
+type Query {
+  rockets: [Rocket]!       @csdl_resolve(graph: ROCKETS)
+  astronauts: [Astronaut]! @csdl_resolve(graph: ASTRONAUTS)
+}
+```
+
+### `@csdl_error`
+
+```graphql
+#::[`@csdl_error` declaration]
+directive @csdl_error(
+  graphs: [csdl_Graph!],
+  message: String)
+    on OBJECT
+     | INTERFACE
+     | UNION
```

Comment:
What does it mean for an error to occur on an entire type, interface, or union? Does this have implications on fields which resolve to any of these (type, interface, union w/errors)?